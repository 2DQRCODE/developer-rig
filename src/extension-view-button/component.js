import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className="circle-button" onClick={this.props.onClick}>
        <p>+</p>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
