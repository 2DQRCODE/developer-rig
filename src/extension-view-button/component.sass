.circle-button
  border-width: thick
  border-radius: 50%
  width: 50px
  height: 50px
  margin: 0 auto
  margin-top: 10px
  border: 2px solid #DAD8DE
  position: relative
  font-weight: bold
  display: table
  cursor: pointer

.circle-button p
  text-align: center
  vertical-align: middle
  display: table-cell

