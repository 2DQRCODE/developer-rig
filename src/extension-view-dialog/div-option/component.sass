@import '../component.sass'
$dimension: 70px

.extension-container__block
  @extend .container__block

  .block__extension-type-label
    visibility: hidden
    position: absolute
    cursor: pointer

  .extension-type-label__extension-type-text--color-black
    @extend .extension-type-text

  .extension-type-label__extension-type-text--color-purple
    @extend .extension-type-text
    color: #7D55C7

.extension-type-box
  margin-top: 10px
  margin-right: 12px
  height: $dimension
  width: $dimension
  border-radius: 5px
  border: 2px solid black
  display: table-cell
  vertical-align: middle
  text-align: center

.extension-type-text
    margin: 3px
    font-size: 10px
    text-align: center
    width: $dimension
