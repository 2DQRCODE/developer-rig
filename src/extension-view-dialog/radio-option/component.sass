.option-label
  clear: left
  font-size: 12px
  margin-right: 10px
