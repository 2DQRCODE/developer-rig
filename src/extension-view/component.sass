.view
  border: 1px solid #D3D0D6
  margin-left: auto
  margin-right: auto

.view__wrapper
  padding: 2em
  padding-top: 0
  border-left: 1px solid #DAD8DE
  border-right: 1px solid #DAD8DE
  border-bottom: 1px solid #DAD8DE
  flex-grow: 1
  flex-basis: auto
  display: flex
  flex-direction: column

.view__header
  padding: 1em
  height: 20px

.view__close_button
  float: right
  padding-left: .8em
  color: #511798
  cursor: pointer

.view__descriptor
  float: left
  padding-right: .8em
  font-size: .75em
  color: #A4A4A4

.rig-frame
  flex-grow: 1
