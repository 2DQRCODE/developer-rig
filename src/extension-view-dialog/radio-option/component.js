import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class RadioOption extends Component {
  render() {
    return (
      <div>
        <input type="radio" name={this.props.name} value={this.props.value}  onChange={this.props.onChange} defaultChecked={this.props.checked}/>
        <label className="option-label"> {this.props.value} </label>
      </div>
    )
  }
}

RadioOption.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool
}
