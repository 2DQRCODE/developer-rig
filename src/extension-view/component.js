import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
import { ExtensionFrame } from '../extension-frame';
import { IdentityOptions } from '../constants/identity-options';
import { ViewerTypes } from '../constants/viewer_types';
import closeButton from '../img/close_icon.png';
const { ExtensionAnchor, ExtensionMode } = window['extension-coordinator'];

export class ExtensionView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      mousedOver: false,
    };
  }

  mouseEnter() {
    this.setState({
      mousedOver: true,
    });
  }

  mouseLeave() {
    this.setState({
      mousedOver: false
    });
  }

  renderLinkedOrUnlinked() {
    return this.props.linked ? IdentityOptions.Linked : IdentityOptions.Unlinked;
  }

  render() {
    const extensionProps = {}
    switch(this.props.type) {
      case ExtensionAnchor.Panel:
        extensionProps.viewStyles = {
          width: "320px",
        };
        break;
      case ExtensionAnchor.Overlay:
        extensionProps.viewStyles = {
          width: this.props.overlaySize.width,
          height: this.props.overlaySize.height
        };
        break;
      case ExtensionMode.Config:
        extensionProps.viewStyles = {
          width: "100%",
          height: "700px"
        };
        extensionProps.viewWrapperStyles = {
          overflow: "auto",
          height: "70vh",
        };
        break;
      default:
        break;
    }

    return (

      <div
        className={'view__wrapper'}
        onMouseEnter={() => { this.mouseEnter() }}
        onMouseLeave={() => { this.mouseLeave() }}
        style={extensionProps.viewWrapperStyles}>
        <div
          className={'view__header'}>
          {(this.props.deleteViewHandler !== undefined && this.state.mousedOver) &&
            (<div className={'view__close_button'}
              onClick={() => { this.props.deleteViewHandler(this.props.id) }}>
            <img
              alt='Close'
              src={closeButton}
            />
            </div>)
          }
          <div className={'view__descriptor'}>
            { this.props.role }
          </div>
          <div className={'view__descriptor'}>
            {(this.props.role === ViewerTypes.LoggedIn) ?
              this.renderLinkedOrUnlinked() : null}
          </div>
        </div>
        <div
          className="view"
          ref={this._boundIframeHostRefHandler}
          style={extensionProps.viewStyles}>
          <ExtensionFrame
            iframeClassName={`rig-frame frameid-${this.props.id}`}
            extension={this.props.extension}
            type={this.props.type}
            mode={this.props.mode}
          />
        </div>
      </div>
    );
  }
}

ExtensionView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  role: PropTypes.string,
  overlaySize: PropTypes.object,
};
