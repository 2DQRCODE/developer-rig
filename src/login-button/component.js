import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class LoginButton extends Component {
  render() {
    return (
      <div className="login-button">
        <button
          onClick={this.props.loginHandler}>Log In</button>
      </div>
    );
  }
}

LoginButton.propTypes = {
  loginHandler: PropTypes.func.isRequired
};
