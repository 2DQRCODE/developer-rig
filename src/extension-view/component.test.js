import React from 'react';
import { shallow } from 'enzyme';
import { ExtensionView } from './component';

describe('<ExtensionView />', () => {
  const extension = {
    authorName: 'test',
    id: 'id',
    description: 'description',
    iconUrl: 'icon_url',
    name: 'name',
    requestIdentity: false,
    sku: 'sku',
    state: 'state',
    summary: 'summary',
    token: 'token',
    vendorCode: 'vendorCode',
    version: 'version',
    views: {},
    whitelistedConfigUrls: 'foo',
    whitelistedPanelUrls: 'bar',
    channelId: 'channelId',
  };
  const iframeClassName = 'rig-frame frameid-0';

  describe('when a panel type', () => {
    const panelType = 'panel';
    const role = 'Broadcaster';
    const mode = 'viewer';
    const linked = false;
    const component = shallow(
      <ExtensionView
        id={'0'}
        extension={extension}
        type={panelType}
        mode={mode}
        role={role}
        linked={linked}
        deleteViewHandler={() => { }}/>
    );

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });


  describe('when a panel type', () => {
    const panelType = 'video_overlay';
    const role = 'Broadcaster';
    const mode = 'viewer';
    const linked = false;
    const component = shallow(
      <ExtensionView
        id={'0'}
        extension={extension}
        type={panelType}
        mode={mode}
        role={role}
        linked={linked}
        overlaySize={{
          width: '1px',
          height: '1px',
        }}
        deleteViewHandler={() => { }}/>
    );

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
