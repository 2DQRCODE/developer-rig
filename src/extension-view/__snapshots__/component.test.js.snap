// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionView /> when a panel type renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionView
    deleteViewHandler={[Function]}
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "version",
            "views": Object {},
            "whitelistedConfigUrls": "foo",
            "whitelistedPanelUrls": "bar",
          }
    }
    id="0"
    linked={false}
    mode="viewer"
    role="Broadcaster"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="view__header"
>
          <div
                    className="view__descriptor"
          >
                    Broadcaster
          </div>
          <div
                    className="view__descriptor"
          />
</div>,
        <div
          className="view"
          style={
                    Object {
                              "height": undefined,
                              "width": "320px",
                            }
          }
>
          <ExtensionFrame
                    extension={
                              Object {
                                        "authorName": "test",
                                        "channelId": "channelId",
                                        "description": "description",
                                        "iconUrl": "icon_url",
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentity": false,
                                        "sku": "sku",
                                        "state": "state",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "version",
                                        "views": Object {},
                                        "whitelistedConfigUrls": "foo",
                                        "whitelistedPanelUrls": "bar",
                                      }
                    }
                    iframeClassName="rig-frame frameid-0"
                    mode="viewer"
                    type="panel"
          />
</div>,
      ],
      "className": "view__wrapper",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="view__descriptor"
>
              Broadcaster
</div>,
            <div
              className="view__descriptor"
/>,
          ],
          "className": "view__header",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Broadcaster",
              "className": "view__descriptor",
            },
            "ref": null,
            "rendered": "Broadcaster",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "view__descriptor",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionFrame
            extension={
                        Object {
                                    "authorName": "test",
                                    "channelId": "channelId",
                                    "description": "description",
                                    "iconUrl": "icon_url",
                                    "id": "id",
                                    "name": "name",
                                    "requestIdentity": false,
                                    "sku": "sku",
                                    "state": "state",
                                    "summary": "summary",
                                    "token": "token",
                                    "vendorCode": "vendorCode",
                                    "version": "version",
                                    "views": Object {},
                                    "whitelistedConfigUrls": "foo",
                                    "whitelistedPanelUrls": "bar",
                                  }
            }
            iframeClassName="rig-frame frameid-0"
            mode="viewer"
            type="panel"
/>,
          "className": "view",
          "style": Object {
            "height": undefined,
            "width": "320px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "extension": Object {
              "authorName": "test",
              "channelId": "channelId",
              "description": "description",
              "iconUrl": "icon_url",
              "id": "id",
              "name": "name",
              "requestIdentity": false,
              "sku": "sku",
              "state": "state",
              "summary": "summary",
              "token": "token",
              "vendorCode": "vendorCode",
              "version": "version",
              "views": Object {},
              "whitelistedConfigUrls": "foo",
              "whitelistedPanelUrls": "bar",
            },
            "iframeClassName": "rig-frame frameid-0",
            "mode": "viewer",
            "type": "panel",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="view__header"
>
            <div
                        className="view__descriptor"
            >
                        Broadcaster
            </div>
            <div
                        className="view__descriptor"
            />
</div>,
          <div
            className="view"
            style={
                        Object {
                                    "height": undefined,
                                    "width": "320px",
                                  }
            }
>
            <ExtensionFrame
                        extension={
                                    Object {
                                                "authorName": "test",
                                                "channelId": "channelId",
                                                "description": "description",
                                                "iconUrl": "icon_url",
                                                "id": "id",
                                                "name": "name",
                                                "requestIdentity": false,
                                                "sku": "sku",
                                                "state": "state",
                                                "summary": "summary",
                                                "token": "token",
                                                "vendorCode": "vendorCode",
                                                "version": "version",
                                                "views": Object {},
                                                "whitelistedConfigUrls": "foo",
                                                "whitelistedPanelUrls": "bar",
                                              }
                        }
                        iframeClassName="rig-frame frameid-0"
                        mode="viewer"
                        type="panel"
            />
</div>,
        ],
        "className": "view__wrapper",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <div
                className="view__descriptor"
>
                Broadcaster
</div>,
              <div
                className="view__descriptor"
/>,
            ],
            "className": "view__header",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Broadcaster",
                "className": "view__descriptor",
              },
              "ref": null,
              "rendered": "Broadcaster",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "view__descriptor",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExtensionFrame
              extension={
                            Object {
                                          "authorName": "test",
                                          "channelId": "channelId",
                                          "description": "description",
                                          "iconUrl": "icon_url",
                                          "id": "id",
                                          "name": "name",
                                          "requestIdentity": false,
                                          "sku": "sku",
                                          "state": "state",
                                          "summary": "summary",
                                          "token": "token",
                                          "vendorCode": "vendorCode",
                                          "version": "version",
                                          "views": Object {},
                                          "whitelistedConfigUrls": "foo",
                                          "whitelistedPanelUrls": "bar",
                                        }
              }
              iframeClassName="rig-frame frameid-0"
              mode="viewer"
              type="panel"
/>,
            "className": "view",
            "style": Object {
              "height": undefined,
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "extension": Object {
                "authorName": "test",
                "channelId": "channelId",
                "description": "description",
                "iconUrl": "icon_url",
                "id": "id",
                "name": "name",
                "requestIdentity": false,
                "sku": "sku",
                "state": "state",
                "summary": "summary",
                "token": "token",
                "vendorCode": "vendorCode",
                "version": "version",
                "views": Object {},
                "whitelistedConfigUrls": "foo",
                "whitelistedPanelUrls": "bar",
              },
              "iframeClassName": "rig-frame frameid-0",
              "mode": "viewer",
              "type": "panel",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionView /> when a panel type renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionView
    deleteViewHandler={[Function]}
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "version",
            "views": Object {},
            "whitelistedConfigUrls": "foo",
            "whitelistedPanelUrls": "bar",
          }
    }
    id="0"
    linked={false}
    mode="viewer"
    overlaySize={
        Object {
            "height": "1px",
            "width": "1px",
          }
    }
    role="Broadcaster"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="view__header"
>
          <div
                    className="view__descriptor"
          >
                    Broadcaster
          </div>
          <div
                    className="view__descriptor"
          />
</div>,
        <div
          className="view"
          style={
                    Object {
                              "height": "1px",
                              "width": "1px",
                            }
          }
>
          <ExtensionFrame
                    extension={
                              Object {
                                        "authorName": "test",
                                        "channelId": "channelId",
                                        "description": "description",
                                        "iconUrl": "icon_url",
                                        "id": "id",
                                        "name": "name",
                                        "requestIdentity": false,
                                        "sku": "sku",
                                        "state": "state",
                                        "summary": "summary",
                                        "token": "token",
                                        "vendorCode": "vendorCode",
                                        "version": "version",
                                        "views": Object {},
                                        "whitelistedConfigUrls": "foo",
                                        "whitelistedPanelUrls": "bar",
                                      }
                    }
                    iframeClassName="rig-frame frameid-0"
                    mode="viewer"
                    type="video_overlay"
          />
</div>,
      ],
      "className": "view__wrapper",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="view__descriptor"
>
              Broadcaster
</div>,
            <div
              className="view__descriptor"
/>,
          ],
          "className": "view__header",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Broadcaster",
              "className": "view__descriptor",
            },
            "ref": null,
            "rendered": "Broadcaster",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "view__descriptor",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionFrame
            extension={
                        Object {
                                    "authorName": "test",
                                    "channelId": "channelId",
                                    "description": "description",
                                    "iconUrl": "icon_url",
                                    "id": "id",
                                    "name": "name",
                                    "requestIdentity": false,
                                    "sku": "sku",
                                    "state": "state",
                                    "summary": "summary",
                                    "token": "token",
                                    "vendorCode": "vendorCode",
                                    "version": "version",
                                    "views": Object {},
                                    "whitelistedConfigUrls": "foo",
                                    "whitelistedPanelUrls": "bar",
                                  }
            }
            iframeClassName="rig-frame frameid-0"
            mode="viewer"
            type="video_overlay"
/>,
          "className": "view",
          "style": Object {
            "height": "1px",
            "width": "1px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "extension": Object {
              "authorName": "test",
              "channelId": "channelId",
              "description": "description",
              "iconUrl": "icon_url",
              "id": "id",
              "name": "name",
              "requestIdentity": false,
              "sku": "sku",
              "state": "state",
              "summary": "summary",
              "token": "token",
              "vendorCode": "vendorCode",
              "version": "version",
              "views": Object {},
              "whitelistedConfigUrls": "foo",
              "whitelistedPanelUrls": "bar",
            },
            "iframeClassName": "rig-frame frameid-0",
            "mode": "viewer",
            "type": "video_overlay",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="view__header"
>
            <div
                        className="view__descriptor"
            >
                        Broadcaster
            </div>
            <div
                        className="view__descriptor"
            />
</div>,
          <div
            className="view"
            style={
                        Object {
                                    "height": "1px",
                                    "width": "1px",
                                  }
            }
>
            <ExtensionFrame
                        extension={
                                    Object {
                                                "authorName": "test",
                                                "channelId": "channelId",
                                                "description": "description",
                                                "iconUrl": "icon_url",
                                                "id": "id",
                                                "name": "name",
                                                "requestIdentity": false,
                                                "sku": "sku",
                                                "state": "state",
                                                "summary": "summary",
                                                "token": "token",
                                                "vendorCode": "vendorCode",
                                                "version": "version",
                                                "views": Object {},
                                                "whitelistedConfigUrls": "foo",
                                                "whitelistedPanelUrls": "bar",
                                              }
                        }
                        iframeClassName="rig-frame frameid-0"
                        mode="viewer"
                        type="video_overlay"
            />
</div>,
        ],
        "className": "view__wrapper",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <div
                className="view__descriptor"
>
                Broadcaster
</div>,
              <div
                className="view__descriptor"
/>,
            ],
            "className": "view__header",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Broadcaster",
                "className": "view__descriptor",
              },
              "ref": null,
              "rendered": "Broadcaster",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "view__descriptor",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExtensionFrame
              extension={
                            Object {
                                          "authorName": "test",
                                          "channelId": "channelId",
                                          "description": "description",
                                          "iconUrl": "icon_url",
                                          "id": "id",
                                          "name": "name",
                                          "requestIdentity": false,
                                          "sku": "sku",
                                          "state": "state",
                                          "summary": "summary",
                                          "token": "token",
                                          "vendorCode": "vendorCode",
                                          "version": "version",
                                          "views": Object {},
                                          "whitelistedConfigUrls": "foo",
                                          "whitelistedPanelUrls": "bar",
                                        }
              }
              iframeClassName="rig-frame frameid-0"
              mode="viewer"
              type="video_overlay"
/>,
            "className": "view",
            "style": Object {
              "height": "1px",
              "width": "1px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "extension": Object {
                "authorName": "test",
                "channelId": "channelId",
                "description": "description",
                "iconUrl": "icon_url",
                "id": "id",
                "name": "name",
                "requestIdentity": false,
                "sku": "sku",
                "state": "state",
                "summary": "summary",
                "token": "token",
                "vendorCode": "vendorCode",
                "version": "version",
                "views": Object {},
                "whitelistedConfigUrls": "foo",
                "whitelistedPanelUrls": "bar",
              },
              "iframeClassName": "rig-frame frameid-0",
              "mode": "viewer",
              "type": "video_overlay",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
